src/main/scala/org/ardlema/solutions/aggregating/AggregateTopologyBuilder.scala

```diff
@@ -25,26 +27,23 @@ object AggregateTopologyBuilder {
                      inputTopic: String,
                      outputTopic: String): Topology = {
 
-    val builder = new StreamsBuilder()
-    val initialStream = builder.stream(inputTopic, Consumed.`with`(Serdes.String(), getAvroSongSerde(schemaRegistryHost, schemaRegistryPort)))
 
-    val artistCount = new KeyValueMapper[String, Song, KeyValue[String, JLong]]() {
+    implicit val stringSerde: Serde[String] = Serdes.String
+    implicit val longSerde: Serde[Long] = Serdes.Long
+    implicit val avroSongSerde: SpecificAvroSerde[Song] = getAvroSongSerde(schemaRegistryHost, schemaRegistryPort)
+
+
+    val builder: StreamsBuilder = new StreamsBuilder()
+    val initialStream: KStream[String, Song] = builder.stream(inputTopic)
+
+    val songsMappedByArtistStream: KStream[String, Long] = initialStream.map((_, song) => (song.getArtist.toString, 1L))
 
-      @Override
-      def apply(key: String, song: Song): KeyValue[String, JLong] = {
-        new KeyValue[String, JLong](song.getArtist.toString, 1L)
-      }
-    }
+    val songsGroupByArtistStream: KGroupedStream[String, Long] = songsMappedByArtistStream.groupByKey
 
+    val songsByArtistStream: KTable[String, Long] = songsGroupByArtistStream.count()
```

Comment:
@eambrosio would you mind changing the name of this val? It is a KTable so it should not contain Stream in its name